pkg/controller/machinepool/awsactuator.go

```diff
@@ -227,9 +228,30 @@ func (a *AWSActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *hi
 		}
 	}
 
+	var vpcID string
+	if metav1.HasAnnotation(pool.ObjectMeta, constants.ExtraWorkerSecurityGroupAnnotation) {
+		if len(pool.Spec.Platform.AWS.Subnets) == 0 {
```

Comment:
1. I'm curious why you decided to do this out here rather than within the `updateProviderConfig` func where you've already got a branch for the annotation.
2. Can the discovery portion of this be factored out into a func that can be reused between this and the installmanager?

Reply:
1. I have this outside `updateProviderConfig` to avoid making an API call per machine set.
2. Yes.

Reply:
> 1. I have this outside `updateProviderConfig` to avoid making an API call per machine set.

D'oh, I see that now, thanks.

> 2. Yes.

:+1: